// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message DnsResourceRecordSet {
  string dns_name = 1;
  string dns_type = 2;
  int64 ttl = 3;
  repeated string target = 4;
  string managed_zone = 5;
  string project = 6;
}

message ApplyDnsResourceRecordSetRequest {
  DnsResourceRecordSet resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteDnsResourceRecordSetRequest {
  string service_account_file = 1;
  DnsResourceRecordSet resource = 2;
}

message ListDnsResourceRecordSetRequest {
  string service_account_file = 1;
  string Project = 2;
  string ManagedZone = 3;
}

message ListDnsResourceRecordSetResponse {
  repeated DnsResourceRecordSet items = 1;
}

service DnsResourceRecordSetService {
  rpc ApplyDnsResourceRecordSet(ApplyDnsResourceRecordSetRequest) returns (DnsResourceRecordSet);
  rpc DeleteDnsResourceRecordSet(DeleteDnsResourceRecordSetRequest) returns (google.protobuf.Empty);
  rpc ListDnsResourceRecordSet(ListDnsResourceRecordSetRequest) returns (ListDnsResourceRecordSetResponse);
}
